quiz.py

```diff
@@ -0,0 +1,344 @@
+import os
+import random as r
+from datetime import datetime
+from time import sleep
+
+import matplotlib
+import numpy as np
+from PIL import Image, ImageDraw
+from matplotlib import patheffects
+from matplotlib import pyplot as plt
+from matplotlib.cbook import get_sample_data
+from matplotlib.offsetbox import (OffsetImage, AnnotationBbox)
+from telegram import Poll, ParseMode, Update
+from telegram.ext import CallbackContext
+from telegram.utils.helpers import mention_html
+
+from constants import group_ids
+from helpers.logger import logger
+from helpers.namer import get_nick
+from online import quiz_scraper
+
+quizzes = []
+cwd = os.getcwd()
+
+
+def send_quiz(context: CallbackContext) -> None:
+    """Sends 5 quizzes to target chat (12B for now). Also sets a timer for 24 hours for quiz expiry (using jobs)."""
+
+    global quizzes
+
+    right_now = datetime.now()  # returns: Datetime obj
+    if 'last_quiz' not in context.bot_data:
+        context.bot_data['last_quiz'] = right_now
+
+    diff = right_now - context.bot_data['last_quiz']
+    print(diff)
+    if diff.days < 7:
+        print("Not enough days for next quiz!")
+        return
+
+    starts = ["See I'm keeping one quizizz now okay. You have one day to finish. For boards ok. I want everyone to do "
+              "it that's it.", "I have kept one quizizz now. I expect something okay.",
+              "Because of the bad like you say situation I have kept this online quizizz now. Do fast okay.",
+              "I'm sending these 5 questions now like. I want it to be done by tomorrow okay? Fast fast"]
+
+    context.bot.send_message(chat_id=group_ids['12b'], text=r.choice(starts))
+
+    # Get our questions, choices and answers from the web-
+    while True:
+        try:
+            questions, choices, answers = quiz_scraper.a_quiz()
+            break
+        except TypeError:  # If we get None (due to error), retry.
+            pass
+
+    # Support sending quiz to 12B only for now-
+    # TODO: Change this back to 12B
+    for question, choice, answer in zip(questions, choices, answers):
+        quiz = context.bot.send_poll(chat_id=group_ids['12b'], question=question, options=choice, is_anonymous=False,
+                                     type=Poll.QUIZ, correct_option_id=answer, is_closed=False)
+        quizzes.append(quiz)
+
+    logger(message=f"The 5 quizzes were just sent to 12B successfully.")
```

Comment:
Ain't it just _one_ 5 question quiz?

Reply:
Technically, Telegram counts it as 5 separate quizzes